sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeedFactory.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.ChangeFeed.Models;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.Blobs.ChangeFeed
+{
+    internal class ChangeFeedFactory
+    {
+        private readonly SegmentFactory _segmentFactory;
+
+        public ChangeFeedFactory()
+        {
+            _segmentFactory = new SegmentFactory(
+                new ShardFactory(
+                    new ChunkFactory(
+                        new LazyLoadingBlobStreamFactory(),
+                        new AvroReaderFactory())));
+        }
+
+        public ChangeFeedFactory(SegmentFactory segmentFactory)
+        {
+            _segmentFactory = segmentFactory;
+        }
+
+        public async Task<ChangeFeed> BuildChangeFeed(
+            bool async,
+            BlobServiceClient blobServiceClient,
+            DateTimeOffset? startTime = default,
+            DateTimeOffset? endTime = default,
+            string continuation = default)
+        {
+            BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(Constants.ChangeFeed.ChangeFeedContainerName);
+            DateTimeOffset lastConsumable;
+            Queue<string> years;
+            Queue<string> segments;
+            ChangeFeedCursor cursor = null;
+
+            // Create cursor
+            if (continuation != null)
+            {
+                cursor = JsonSerializer.Deserialize<ChangeFeedCursor>(continuation);
+                ValidateCursor(containerClient, cursor);
+                startTime = cursor.CurrentSegmentCursor.SegmentTime;
+                endTime = cursor.EndTime;
+            }
+            // Round start and end time if we are not using the cursor.
+            else
+            {
+                startTime = startTime.RoundDownToNearestHour();
+                endTime = endTime.RoundUpToNearestHour();
+            }
+
+            // Check if Change Feed has been abled for this account.
+            bool changeFeedContainerExists;
+
+            if (async)
+            {
+                changeFeedContainerExists = await containerClient.ExistsAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                changeFeedContainerExists = containerClient.Exists();
+            }
+
+            if (!changeFeedContainerExists)
+            {
+                throw new ArgumentException("Change Feed hasn't been enabled on this account, or is currently being enabled.");
```

Comment:
or is currently being **disabled**

Reply:
Nope, this is by design.  It takes some time to enable change feed for an account.